cuckoo-miner/src/miner/miner.rs

```diff
@@ -94,29 +94,29 @@ impl CuckooMiner {
 		let control_ctx = SolverCtxWrapper(NonNull::new(ctx).unwrap());
 
 		let stop_fn = solver.lib.get_stop_solver_instance();
-		let sleep_dur = time::Duration::from_millis(100);
-		// monitor whether to send a stop signal to the solver, which should
-		// end the current solve attempt below
-		let stop_handle = thread::spawn(move || loop {
-			let ctx_ptr = control_ctx.0.as_ptr();
+
+		let mut iter_count = 0;
+		let mut paused = true;
+		let ctx_ptr = control_ctx.0.as_ptr();
+		'solver_loop: loop {
+			// monitor whether to send a stop signal to the solver, which should
+			// end the current solve attempt below
```

Comment:
It seems to me that we should not put this code into the same thread as solver, we will never have a chance to stop the current attempt because we check stop/pause messages before running the solver.

Reply:
👍   agree! and to revert this mistake thread merging, I write a new PR https://github.com/mimblewimble/grin-miner/pull/84